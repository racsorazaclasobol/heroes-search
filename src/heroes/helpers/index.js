export * from './GetHeroByPublisher'
export * from './GetHeroById'
export * from './GetHeroByName'