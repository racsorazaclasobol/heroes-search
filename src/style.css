body {
    padding: 15px;
}

NavBar { 
    padding: 15px;
}