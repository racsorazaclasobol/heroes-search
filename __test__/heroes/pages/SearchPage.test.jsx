import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({ 
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}))


describe('Tests on <SearchPage />', () => { 

    beforeAll( () => jest.clearAllMocks() );
    
    test('should show default values correctly ', () => { 

        const { container } = render( 
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
         );

         expect( container ).toMatchSnapshot();
        //  screen.debug();

     })

    test('debe de mostrar a Batman y el input debe estar con el valor de Batman, a traves del queryString en la url ', () => { 

        render( 
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
         );

        // screen.debug();

        const inputValue = screen.getByRole( 'textbox' );
        expect( inputValue.value ).toBe( 'batman' );

        const imgValue = screen.getByRole( 'img' );
        expect( imgValue.src ).toContain('/assets/heroes/dc-batman.jpg');

        const divSearch = screen.getByTestId('divSearch');
        expect(divSearch.style.display).toBe('none');

    })

    test('should show an error if not found a hero', () => { 

        render( 
            <MemoryRouter initialEntries={['/search?q=asdasdasdaas']}>
                <SearchPage />
            </MemoryRouter>
         );

         const divSearch = screen.getByTestId('divError');
         expect(divSearch.style.display).toBe('');

    })

    test('should show an error if not found a hero', () => { 

        render( 
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
         );

         const inputValue = screen.getByRole( 'textbox' );

         fireEvent.change(inputValue, {target: {name: 'searchText', value:'superman'}})

         const form = screen.getByTestId( 'form' );
         fireEvent.submit(form);

         expect( mockedUseNavigate ).toHaveBeenCalledWith('?q=superman')
    
    })

 })