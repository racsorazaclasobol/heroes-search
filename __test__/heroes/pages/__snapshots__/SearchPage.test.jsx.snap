// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <SearchPage /> should show default values correctly  1`] = `
<div>
  <h1>
    Search Page
  </h1>
  <br />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching...
      </h4>
      <hr />
      <form
        data-testid="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-4"
        >
           Search 
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
        data-testid="divSearch"
      >
        Search a hero.
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        data-testid="divError"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
